pkg/klusterlet/clusterca/clusterca_controller.go

```diff
@@ -0,0 +1,134 @@
+	managedClusterInfo := &clusterv1beta1.ManagedClusterInfo{}
+	err := r.Client.Get(ctx, req.NamespacedName, managedClusterInfo)
+	if err != nil {
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+	newStatus := managedClusterInfo.DeepCopy().Status
+	if newStatus.DistributionInfo.Type != clusterv1beta1.DistributionTypeOCP {
+		return ctrl.Result{}, nil
+	}
+
+	//get ocp apiserver url
+	kubeAPIServer, err := utils.GetKubeAPIServerAddress(r.OpenshiftClient)
+	if err != nil {
+		klog.Errorf("Failed to get kube Apiserver. err:%v", err)
+		return ctrl.Result{}, err
+	}
+
+	//get ocp ca
+	clusterca := r.getClusterCA(kubeAPIServer)
+	if len(clusterca) <= 0 {
+		klog.Errorf("Failed to get clusterca.")
+		return ctrl.Result{}, fmt.Errorf("failed to get clusterca")
+	}
+
+	klog.V(5).Infof("kubeapiserver: %v, clusterca:%v", kubeAPIServer, clusterca)
+
+	var needUpdate = false
+	for i, clientConfig := range newStatus.DistributionInfo.OCP.ManagedClusterClientConfigs {
+		if clientConfig.URL != kubeAPIServer {
+			continue
+		}
+		if reflect.DeepEqual(clusterca, clientConfig.CABundle) {
+			return ctrl.Result{RequeueAfter: UpdateInterval}, nil
+		}
+		needUpdate = true
+		newStatus.DistributionInfo.OCP.ManagedClusterClientConfigs[i].CABundle = clusterca
+	}
+
+	if !needUpdate {
+		newConfig := clusterv1beta1.ClientConfig{
+			URL:      kubeAPIServer,
+			CABundle: clusterca,
+		}
+		newStatus.DistributionInfo.OCP.ManagedClusterClientConfigs = append(newStatus.DistributionInfo.OCP.ManagedClusterClientConfigs, newConfig)
+	}
+	managedClusterInfo.Status = newStatus
+	err = r.Client.Status().Update(ctx, managedClusterInfo, &client.UpdateOptions{})
+	if err != nil {
+		klog.Errorf("Failed to update managedclusterinfo status, err:%v", err)
+		return ctrl.Result{}, err
+	}
+
+	return ctrl.Result{RequeueAfter: UpdateInterval}, nil
+}
+
+func (r *ClusterCaReconciler) getClusterCA(kubeAPIServer string) []byte {
+	//Get ca from apiserver
+	certData, err := utils.GetCAFromApiserver(r.OpenshiftClient, r.KubeClient, kubeAPIServer)
+	if err == nil && len(certData) > 0 {
+		return certData
+	}
+	klog.V(3).Infof("Failed to get ca from apiserver, error:%v", err)
+
+	//Get ca from configmap in kube-public namespace
+	certData, err = utils.GetCAFromConfigMap(r.KubeClient)
+	if err == nil && len(certData) > 0 {
+		return certData
+	}
+
+	klog.V(3).Infof("Failed to get ca from kubepublic namespace configmap, error:%v", err)
+
+	//Fallback to service account token ca.crt
+	certData, err = utils.GetCAFromServiceAccount(r.KubeClient)
+	if err == nil && len(certData) > 0 {
+		// check if it's roks
+		// if it's ocp && it's on ibm cloud, we treat it as roks
+		isROKS, err := utils.CheckIsIBMCloud(r.KubeClient)
+		if err != nil {
+			return certData
+		}
+		if isROKS {
+			// ROKS should have a certificate that is signed by trusted CA
+			certs, err := utils.GetValidCertificatesFromURL(kubeAPIServer, nil)
```

Comment:
why do we need this?

Reply:
IBM cloud have public ca, so it need to be checked here.
Actually, we got the logic from import controller part. 
https://github.com/open-cluster-management/managedcluster-import-controller/blob/33cf2014906d2c1d3087ccbee6d1dd0b3101098e/pkg/controller/managedcluster/import_yamls.go#L240

Reply:
if it needs to be set to nil, why try to get it from request?